### ISSUE TITLE
a11y: Fix role missing for action menu component

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12816]
a11y issue: 89

Interactive role is not defined,

The role such as link and button is not defined for the interactive elements present in the main content area resulting in the interactive elements being announced as plain text by the screen reader.

Multiple occurrences need to be fixed - but should all be fixed by fixing the Action menu component.

Account and API Keys
 - API Keys section
  - View in API and Delete (that becomes available on activating "More Actions" image button)

Extensions
- Manage Repositories
- Add Rancher Repositories
- Manage Extension Catalogs
- Developer Load

Example:

![image](https://github.com/user-attachments/assets/1a4ff48e-fb8e-4131-ae26-f47cf91c5b20)

Suggested remediation:

Apply the following changes:
• Code the "View in API" interactive element using <a> tag and include the href attribute with appropriate value in the source code.
• Code the "Delete" interactive element using <button> tag. 
• Modify the scripts to ensure that the functionality is accessible for all users.

**Comment by izaac** [comment]
Rancher version: `v2.11-0b081e077e830fa17b86151d21c786fb8d07c6de-head`
UI version: `master 68fb044`
Chrome version: `133.0.6943.141`
Firefox version: `135.0.1`

I validated all the mentioned tables' actions dropdown. They can be navigated with keyboard only and picked by the screen reader. ✅
